### PR TITLE
Submitting Secretaria.el

### PR DESCRIPTION
**Pull request by shackra**
### Brief summary of what the package does

My attempt of an assistant for you in org-mode. This library contains
utilities that enhance your experience with org-mode.

#### Features
  - Reminders
      - All tasks scheduled or that have a deadline set to "today", but
        have no time of the day specified.
      - The current clocked task, every N minutes (10 by default).
      - In case of Emacs crashing, the task clocked in at the moment so
        you don't forget about fixing that.

### Direct link to the package repository

https://bitbucket.org/shackra/secretaria.el

### Your association with the package

I'm the creator and current maintainer.

### Relevant communications with the upstream package maintainer

None.

### Checklist

- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've built and installed the package using the instructions in the [README](https://github.com/melpa/melpa/blob/master/README.md)

### CODE REVIEW

**Comment by syohex** [comment]
- Package dependencies in `secretaria.el` is different from `secretaria-pkg.el`(You can remove pkg file by moving version information into `secretaria.el`)
- No package footer in `secretaria.el`.
- Please use `-` separator instead of `/` for emacs lisp coding convention
- There are too many byte-compile warnings, please check them
- The package uses `string-empty-p` which was introduced and Emacs 24.4 in subr-x.el. But minimum Emacs version is 24.1

**Comment by shackra** [comment]
But I'm using Git on Bitbucket.org, so I am doing the recipe as it says on MELPA's instructions, don't I?

I'll me fixing those other mistakes ASAP

El 10 de octubre de 2016 8:01:27 GMT-06:00, Syohei YOSHIDA <notifications@github.com> escribió:
>syohex commented on this pull request.
>
>
>
>> @@ -0,0 +1 @@
>+(secretaria :fetcher git :url
>"https://bitbucket.org/shackra/secretaria.el")
>
>You should use :fetcher bitbucket instead of git. Could you check other
>recipes which are hosted on bitbucket ?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/melpa/melpa/pull/4307#pullrequestreview-3504443

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

File: recipes/secretaria [hunk 1]
```diff
@@ -0,0 +1 @@
+(secretaria :fetcher git :url "https://bitbucket.org/shackra/secretaria.el")
```
Comment (shackra) [inline recipes/secretaria]: I'm using Git on Bitbucket.org, so as the MELPA documentatio says, I should be using git as fetcher.

**Comment by shackra** [comment]
Most things are fixed now. But regarding the byte compiles warnings, I got none! I don't know if I'm doing something wrong.

File: recipes/secretaria (hunk 1 above)
Comment (shackra) [inline recipes/secretaria]: No problem :D

**Comment by syohex** [comment]
I have sent PR. Please check it. However there are still some byte-compile warnings.
- https://bitbucket.org/shackra/secretaria.el/pull-requests/1

**Comment by syohex** [comment]
I think files(in this case, `~/.secretaria-clocked-task`) which are used from the package should be restored under `user-emacs-directory`. (See `locate-user-emacs-file`)

**Comment by shackra** [comment]
I've move the file to a variable so people can customize it to their liking, the file will be stored inside .emacs.d by default.

I think we are good to go!

**Comment by alphapapa** [comment]
@shackra byte-compile warnings often don't show up when testing your own packages in your own configuration.  Usually you can find them by installing your own package inside the MELPA sandbox (`make sandbox` in the `melpa` repo directory).

**Comment by purcell** [comment]
Thanks. Feedback from me:
- Is it really necessary to depend on org 8.3? I have Emacs HEAD locally, and this dependency causes Emacs to install the separate org snapshot package, while I expect that the bundled Org is sufficient.
- `secretaria--style-get-best-available` should not exist. You're using `alert`, so why would you try to decide for the user what alert style they will most want?
- The package description `An attempt of an assistant for you in org-mode` doesn't read well: I suggest you replace it with something like `A personal assistant based on org-mode`.

The byte-compilation warnings I see are the following:
```
In secretaria-due-appt:
secretaria-appt.el:102:21:Warning: reference to free variable
    ‘secretaria-style-best-available’

In secretaria-today-unknown-time-appt:
secretaria-appt.el:121:31:Warning: reference to free variable
    ‘secretaria-style-best-available’

In secretaria-today-unknown-time-appt-always-remind-me:
secretaria-appt.el:137:9:Warning: assignment to free variable
    ‘secretaria-today-uknown-time-reminder-timer’

In secretaria--style-get-best-available:
secretaria-clock.el:59:19:Warning: reference to free variable
    ‘dbus-compiled-version’
```

**Comment by shackra** [comment]
I've implemented the changes you suggest and revised the byte-compile warnings, it should be
ready now!.

El sábado 15 de octubre del 2016 a las 0224 horas, Steve Purcell escribió:

> Thanks. Feedback from me:
> - Is it really necessary to depend on org 8.3? I have Emacs HEAD locally, and this dependency causes Emacs to install the separate org snapshot package, while I expect that the bundled Org is sufficient.
> - `secretaria--style-get-best-available` should not exist. You're using `alert`, so why would you try to decide for the user what alert style they will most want?
> - The package description `An attempt of an assistant for you in org-mode` doesn't read well: I suggest you replace it with something like `A personal assistant based on org-mode`.
>
> The byte-compilation warnings I see are the following:
> ```
> In secretaria-due-appt:
> secretaria-appt.el:102:21:Warning: reference to free variable
>     ‘secretaria-style-best-available’
>
> In secretaria-today-unknown-time-appt:
> secretaria-appt.el:121:31:Warning: reference to free variable
>     ‘secretaria-style-best-available’
>
> In secretaria-today-unknown-time-appt-always-remind-me:
> secretaria-appt.el:137:9:Warning: assignment to free variable
>     ‘secretaria-today-uknown-time-reminder-timer’
>
> In secretaria--style-get-best-available:
> secretaria-clock.el:59:19:Warning: reference to free variable
>     ‘dbus-compiled-version’
> ```


--
👋 Pax et bonum.
Jorge Araya Navarro
https://es.gravatar.com/shackra

**Comment by purcell** [comment]
Thanks. :-)

**Pull request merged**